### PR TITLE
ci: adds automated Ziglang Stdlib integration test 

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        # --enable-cache persists the binary output to the cache directory and prints the absolute path to the binary.
+        run: |
+          _ZIG_TEST_BINARY_PATH=$(zig test --test-no-exec --enable-cache -target wasm32-wasi --zig-lib-dir ./lib ./lib/std/std.zig)/test.wasm
+          echo "ZIG_BINARY_PATH=${_ZIG_TEST_BINARY_PATH}" >> $GITHUB_OUTPUT
+
+      - name: Run the test binary with wazero CLI
+        run: |
+          go run github.com/tetratelabs/wazero/cmd/wazero@${{ github.sha }} run -hostlogging=filesystem \
+            ${{ steps.zig-build-test-binary.outputs.ZIG_BINARY_PATH }}
+        # TODO: remove before 1.0
```
Comment (codefromthecrypt) [inline .github/workflows/integration.yaml:69]: maybe don't close the zig stdlib tests and leave it open until this passes, then you can say "remove when #1029 is complete" we can rename that issue from "setup tests" to "pass tests" :)

**Comment by mathetake** [comment]
oh wait forgot to pass -mount option

**Comment by mathetake** [comment]
cache is working and almost zero cost to get the test binary
image

**Comment by mathetake** [comment]
cool good to go

**Review by codefromthecrypt — APPROVED** [review]
sweet thanks

**Pull request merged**